pkg/controller/acmechallenges/sync.go

```diff
@@ -92,8 +85,8 @@ func (c *controller) Sync(ctx context.Context, ch *cmacme.Challenge) (err error)
 	// cert-manager has a chance to clean up resources created for the
 	// challenge.
 	if finalizerRequired(ch) {
-		ch, err = addFinalizer(c, ctx, ch)
-		return err
+		ch.Finalizers = append(ch.Finalizers, cmacme.ACMEFinalizer)
```

Comment:
Thinking loudly: so I guess what we will be wanting to do here for SSA is actually to create an 'empty' challenge object where the only field we set is the finalizer. And since we cannot use our existing generation functionality to create these empty objects because there is a risk of taking ownership of defaulted fields, we might need to use unstructured objects. I wonder if the `updater` interface that we added in this PR will fit that model?

Reply:
I don't think the updater interface fits the SSA model, no. But I'm not convinced that we can actually use SSA at the moment and if we do I think we should change all the Sync functions to use applyconigurations or unstructured (as you say).